processor/spanmetricsprocessor/processor.go

```diff
@@ -214,29 +229,43 @@ func (p *processorImp) Capabilities() consumer.Capabilities {
 func (p *processorImp) ConsumeTraces(ctx context.Context, traces pdata.Traces) error {
 	p.aggregateMetrics(traces)
 
-	m := p.buildMetrics()
+	m, err := p.buildMetrics()
+	if err != nil {
+		return err
+	}
 
 	// Firstly, export metrics to avoid being impacted by downstream trace processor errors/latency.
 	if err := p.metricsExporter.ConsumeMetrics(ctx, *m); err != nil {
```

Comment:
Why not consider periodic build and consume metric here?
Each trace goes to export metrics ( for example: remote write、otlp exporter ), will the cost be slightly higher?